### PR TITLE
feat: add query log

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Add tracing target `pgwire_query_log` and `risingwave_frontend_slow_query_log`.
- Support dumping `pgwire_query_log` to `.risingwave/log/query.log` and dumping `risingwave_frontend_slow_query_log` to `.risingwave/log/slow_query.log`.

### CODE REVIEW

File: src/utils/runtime/src/lib.rs [hunk 1]
```diff
@@ -119,8 +125,27 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
 
         fmt_layer.with_filter(filter)
     };
+    let mut layers = vec![fmt_layer.boxed()];
+
+    if ENABLE_QUERY_LOG_FILE {
+        let query_log_path = ".risingwave/log/query.log";
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:131]: I wanted to make it a configurable path, but didn't find a good place. 😇

Using env var in `main_okk` seems ok, but logger init is outside of `main_okk`. 😇

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -25,6 +25,12 @@ use tracing_subscriber::filter;
 use tracing_subscriber::layer::SubscriberExt;
 use tracing_subscriber::prelude::*;
 
+/// Dump logs of all SQLs, i.e., tracing target `pgwire_query_log` to `.risingwave/log/query.log`.
+///
+/// Changing the level of `pgwire` to `TRACE` in `configure_risingwave_targets_fmt` can also turn on
+/// the logs, but without a dedicated file.
+const ENABLE_QUERY_LOG_FILE: bool = false;
```
Comment (BowenXiao1999) [inline src/utils/runtime/src/lib.rs:32]: So this now should be set manually by people?

**Review by BowenXiao1999 — APPROVED** [review]
Is it true that now we can just change the bool flag and then observe the query trace log?

**Comment by BugenZhao** [comment]
What about also changing the target of this?

https://github.com/risingwavelabs/risingwave/blob/c7b2f15a15a94fb6628beb02e879bc64d00aef85/src/frontend/src/session.rs#L667-L677

File: src/utils/runtime/src/lib.rs (hunk 2 above)
Comment (xxchan) [reply src/utils/runtime/src/lib.rs:32]: Yes. Just like modifying log levels.

Comment (xxchan) [reply src/utils/runtime/src/lib.rs:32]: It's not very flexible but it's currently enough for debugging and diagnosis? 🤔

**Review by BugenZhao — APPROVED** [review]

File: src/utils/runtime/src/lib.rs (hunk 2 above)
Comment (fuyufjh) [reply src/utils/runtime/src/lib.rs:32]: I just run into a use case that would be much easier to debug with this query log. Can you please expose it as a config?

**Pull request merged**